pkg/utils/strings.go

```diff
@@ -23,20 +24,32 @@ const (
 	hextable   = "0123456789abcdef"
 )
 
-func RandString(rnd Random, ln int) string {
-	length := ln
-	if val := length % uint64Size; val != 0 {
-		length += uint64Size - val
-	}
+var (
+	randomString     string
+	randomStringOnce sync.Once
+)
+
+func PreallocateRandomString(rnd Random, ln int) {
+	randomStringOnce.Do(func() {
+		length := ln
+		if val := length % uint64Size; val != 0 {
+			length += uint64Size - val
+		}
 
-	binBuff := make([]byte, length)
+		data := make([]byte, ln)
 
-	for i := 0; i < length; i += uint64Size {
-		number := rnd.Int64()
-		for j := 0; j < uint64Size; j++ {
-			binBuff[i+j] = hextable[(number>>j)&0x0f]
+		for i := 0; i < length; i += uint64Size {
+			number := rnd.Uint64()
+			for j := range uint64Size {
+				data[i+j] = hextable[(number>>j)&0x0f]
+			}
 		}
-	}
 
-	return unsafe.String(unsafe.SliceData(binBuff), length)[:ln]
+		randomString = unsafe.String(unsafe.SliceData(data), length)
+	})
+}
+
+func RandString(rnd Random, ln int) string {
+	start := rnd.IntN(len(randomString) - ln - 1)
```

Comment:
`len(randomString)` seem static, buffering this value would bring significant impact?
also, if we know that `ln` will be less than xxx we could hardcode `start` to possible max `ln`  (but I don't know if it's worthwhile optimization)